### ISSUE TITLE
Security concern for terminal access, any plan for auth?

### ISSUE THREAD

**Issue by Ewocker** [weaveworks/scope#2910]
I am new to weave scope. Just got introduced to the great tool a day ago.

I was wondering an issue when I saw the terminal access on each container though. Does it mean that anyone who has access to the UI interface will have full access to each of the containers? I think the doc also mention this point somewhere and suggest to use caddy or weave cloud for better protection, but what if I only want to allow certain group of people to be able to access the control of each container, and rest can only view the general status from basic UI. Neither weave cloud nor caddy or any auth layer on top can solve the problem.

Is it possible to write a plugin for permission control? I have looked at the source code and the plugins tutorial. It seems like it is not designed to limit the functionality but only for extension.
I was looking at the react source code and I would say that if there is no such behavior for disabling the terminal expose for plugins, there is no way that a plugin can achieve this goal. Suppose that writing a plugin can possibly achieve this, how hard do you expect it to be? Is there any plan to make a user permission system on top of the weave scope layer? (I did not see it in the milestone on Github)

In a nutshell, there are ways to protect your services from external attack but no implementation on leveling internal user access. I think it is really important especially for any large scale company deploying multiple services and want to use weave scope for monitoring. If it is part of the plan, I am more than willing to help.

Please correct me if I misunderstand any part of the tool, thanks.

**Comment by rade** [comment]
We will add fine-grained permissions - allowing control over who gets to use what features - to Weave Cloud at some point. There we already have the notion of users, auth, etc.

We are unlikely to add such functionality to scope.

**Comment by calvintam236** [comment]
So it is a no-go for #2814?

**Comment by rade** [comment]
> So it is a no-go for #2814?

Correct, but we would of course accept a PR to implement such functionality.
